### PR TITLE
Added `devcontainer.json`

### PR DESCRIPTION
**Pull request by okineadev**
I added a `devcontainer.json` so that you can **immediately** start development on this project on [**GitHub Codespaces**](https://github.com/features/codespaces)

devcontainer.json contains the pre-installed **brew** and [**Markdownlint**](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) extension for vscode

In order for developers to really start development immediately, I recommend enabling **prebuilds**, this feature pre-installs **brew** and makes a ready system image with all the necessary tools, after which this system image is used in **Codespaces**

Now let's discuss what features and development extensions we can add to **devcontainer**

### CODE REVIEW

**Review by carlocab — COMMENTED** [review]
Thanks. I'm ok with this idea, but I think we should use the configuration from Homebrew/brew. See:

https://github.com/Homebrew/brew/blob/801c737e1bba5b1ffa2362c0c2be1eb265726d17/.devcontainer/devcontainer.json
https://github.com/Homebrew/brew/blob/801c737e1bba5b1ffa2362c0c2be1eb265726d17/.devcontainer/on-create-command.sh

**Comment by okineadev** [comment]
> Thanks. I'm ok with this idea, but I think we should use the configuration from Homebrew/brew. See:
> 
> https://github.com/Homebrew/brew/blob/801c737e1bba5b1ffa2362c0c2be1eb265726d17/.devcontainer/devcontainer.json
> https://github.com/Homebrew/brew/blob/801c737e1bba5b1ffa2362c0c2be1eb265726d17/.devcontainer/on-create-command.sh

Maybe, but it would be good if you applied my extensions and settings that I offered you

**Comment by carlocab** [comment]
> Maybe, but it would be good if you applied my extensions and settings that I offered you

Feel free to open a PR at Homebrew/brew that does this so we can review it properly.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/ubuntu
+{
+  "name": "homebrew-core",
+  // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+  "image": "mcr.microsoft.com/devcontainers/base:jammy",
+  "features": {
+    "ghcr.io/meaningful-ooo/devcontainer-features/homebrew:2": {}
```
Comment (ZhongRuoyu) [inline .devcontainer/devcontainer.json:8]: This is untrusted.

**Review by ZhongRuoyu — COMMENTED** [review]
> Maybe, but it would be good if you applied my extensions and settings that I offered you

If you would like to use these customisations and container features, please add them in your own fork instead.

**Comment by fxcoudert** [comment]
Agree with other reviewers. At this stage, we'll pass on this proposal:
- you can customise your environment in your own fork
- proposals for improvement of the devcontainer should be proposed in Homebrew/brew and reviewed there
- for security reasons, we will not rely on untrusted environments

Thanks for the proposal nonetheless!

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (okineadev) [reply .devcontainer/devcontainer.json:8]: Ok, so I will first make a **PR** with my **vscode** settings and extensions in [Homebrew/brew](https://github.com/Homebrew/brew), and when the **PR** is accepted - then we can use it in **this project** as well

**Comment by okineadev** [comment]
Check out https://github.com/Homebrew/brew/pull/17298

**Comment by MikeMcQuaid** [comment]
If there's a way of telling people who try to create a homebrew-core (and probably homebrew-cask) dev container that they should use the Homebrew/brew one instead: that would be ideal.

**Comment by okineadev** [comment]
@MikeMcQuaid, Do you mean system image?

**Comment by MikeMcQuaid** [comment]
@okineadev No, I mean the Homebrew/brew developer container is also best for Homebrew/homebrew-core development as it contains that repository.

**Comment by okineadev** [comment]
> @okineadev No, I mean the Homebrew/brew developer container is also best for Homebrew/homebrew-core development as it contains that repository.

Got it, thanks

**Pull request closed without merging**